qiskit_qec/operators/base_xp_pauli.py

```diff
@@ -0,0 +1,460 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# Part of the QEC framework
+"""Module for base XP pauli"""
+
+import numbers
+from typing import List, Optional, Union
+
+import numpy as np
+
+# Must be imported as follows to avoid circular import errors
+import qiskit.quantum_info.operators.symplectic.clifford
+from qiskit import QiskitError
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.barrier import Barrier
+from qiskit.circuit.delay import Delay
+from qiskit.circuit.instruction import Instruction
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.operators.mixins import AdjointMixin, MultiplyMixin
+from qiskit_qec.utils import xp_pauli_rep
+
+
+# pylint: disable=no-member
+class BaseXPPauli(BaseOperator, AdjointMixin, MultiplyMixin):
+    r"""Symplectic representation of a list of N-qubit XP operators with phases using
+    numpy arrays for symplectic matrices and phase vectors.
+
+    Base class for XPPauli and XPPauliList.
+    """
+
+    # External string formats used when displaying Pauli's as strings
+    EXTERNAL_TENSOR_ENCODING = xp_pauli_rep.DEFAULT_EXTERNAL_TENSOR_ENCODING
+    EXTERNAL_PHASE_ENCODING = xp_pauli_rep.DEFAULT_EXTERNAL_PHASE_ENCODING
+    EXTERNAL_XP_PAULI_ENCODING = EXTERNAL_PHASE_ENCODING + EXTERNAL_TENSOR_ENCODING
+
+    EXTERNAL_SYNTAX = xp_pauli_rep.PRODUCT_SYNTAX
+    EXTERNAL_QUBIT_ORDER = xp_pauli_rep.DEFAULT_QUBIT_ORDER
+
+    PRINT_PHASE_ENCODING = None
+
+    def __init__(
+        self,
+        matrix: Union[np.ndarray, None] = None,
+        phase_exp: Union[None, np.ndarray, np.integer] = None,
+        order: str = "xz",
+    ) -> None:
+        """A BaseXPPauli object represents a list N-qubit Pauli operators with phases.
+        Numpy arrays are used to represent the symplectic matrix represention of these
+        Paulis. The phases of the Paulis are stored encoded. The phases of the Pauli
```

Comment:
Leave a TODO here as we need to work out the representations to be used for the XP operators. Also the

` (-i)^phase_exp Z^z X^x` 
 
 needs to be updated to XP operators etc.